js/basemap_selector.ts

```diff
@@ -0,0 +1,103 @@
+import type { RenderProps } from "@anywidget/types";
+import { css, html, PropertyValues, TemplateResult } from "lit";
+import { property, query } from "lit/decorators.js";
+
+import { legacyStyles } from "./ipywidgets_styles";
+import { LitWidget } from "./lit_widget";
+import { materialStyles } from "./styles";
+import { loadFonts } from "./utils";
+
+export interface BasemapSelectorModel {
+    basemaps: string[];
+    value: string;
+}
+
+export class BasemapSelector extends LitWidget<
+    BasemapSelectorModel,
+    BasemapSelector
+> {
+    static get componentName(): string {
+        return `basemap-selector`;
+    }
+
+    static styles = [
+        legacyStyles,
+        materialStyles,
+        css`
+            .row-container {
+                align-items: center;
+                display: flex;
+                height: 32px;
+                width: 200px;
+            }
+
+            .row-button {
+                font-size: 14px;
+                height: 26px;
+                margin: 4px;
+                width: 26px;
+            }
+        `,
+    ];
+
+    modelNameToViewName(): Map<
+        keyof BasemapSelectorModel,
+        keyof BasemapSelector
+    > {
+        return new Map([
+            ["basemaps", "basemaps"],
+            ["value", "value"],
+        ]);
+    }
+
+    @property({ type: Array }) basemaps: string[] = [];
+    @property({ type: String }) value: string = "";
+    @query('select') selectElement!: HTMLSelectElement;
+
+    render(): TemplateResult {
+        return html`
+            <div class="row-container">
+                <select class="legacy-select" @change=${this.onChange}>
+                    ${this.basemaps.map((basemap) => html`<option>${basemap}</option>`)}
+                </select>
+                <button
+                    class="legacy-button primary row-button close-button"
+                    @click="${this.onCloseClicked}"
+                >
+                    <span class="close-icon material-symbols-outlined">close</span>
+                </button>
+            </div>`;
+    }
+
+    override update(changedProperties: PropertyValues): void {
+        if (changedProperties.has("value") && this.selectElement) {
```

Comment:
This line implies `selectElement` may not exist, but the declaration `@query('select') selectElement!: HTMLSelectElement;` implies it always exists.

What about declaring `selectElement` as an optional?

Reply:
Good call, done. Technically it shouldn't happen, but I think I ran into it while testing so I added a check here.